src/common/components/addressForm/addressForm.react.tsx

```diff
@@ -0,0 +1,305 @@
+    intAddressLine3: Yup.string()
+      .max(100, 'This field cannot be longer than 100 characters'),
+    intAddressLine4: Yup.string()
+      .max(100, 'This field cannot be longer than 100 characters'),
+    locality: Yup.string()
+      .max(50, 'This field cannot be longer than 100 characters')
+      .required('You must enter a city'),
+    region: Yup.string()
+      .required('You must select a state / region'),
+    postalCode: Yup.string()
+      .test(
+        'is-postal-code',
+        () => 'You must enter a valid US zip code',
+        (value) => value == null || /^\d{5}(?:[-\s]\d{4})?$/.test(value)
+      )
+      .required('You must enter a zip / postal code')
+  });
+
+  const handleAddressChanged = (values: Address) => {
+    onAddressChanged(values);
+  };
+
+  const loadCountries = () => {
+    setLoadingCountriesError(false);
+
+    geographiesService.getCountries()
+      .subscribe((data: GeographiesItem[]) => {
+        const sortedCountries = data.sort(dropdownSortComparator);
+
+        setCountries(sortedCountries);
+
+        const countryContext = countryName && findCountry(sortedCountries, countryName);
+        countryContext && loadRegions(countryContext);
+      },
+      (error: any) => {
+        setLoadingCountriesError(true);
+        $log.error('Error loading countries.', error);
+      });
+  };
+
+  const loadRegions = (countryContext: GeographiesItem) => {
+    setLoadingRegionsError(false);
+
+    geographiesService.getRegions(countryContext)
+      .subscribe((data: GeographiesItem[]) => {
+        const sortedRegions = data.sort(dropdownSortComparator);
+
+        setRegions(sortedRegions);
+      },
+      (error: any) => {
+        setLoadingRegionsError(true);
+        $log.error('Error loading regions.', error);
+      });
+  };
+
+  const findCountry = (countryOptions: GeographiesItem[], countryName?: string): GeographiesItem | undefined => {
+    let foundCountry: GeographiesItem | undefined = undefined;
+    
+    if (countryOptions.length > 0) {
+      foundCountry = find(countryOptions, { name: countryName });   
+    }
+
+    return foundCountry;
+  };
+
+  const refreshRegions = () => {
+    const countryContext = countryName && findCountry(countries, countryName);
+    countryContext && loadRegions(countryContext);
+  }
+
+  return (
+    <Formik
+      initialValues={address}
+      validationSchema={AddressSchema}
+      onSubmit={handleAddressChanged}
+    >
+      {({
+        values,
+        errors,
+        touched,
+        handleChange,
+        handleBlur,
+      }) => (
+        <>
+          <FormikAutoSave debounceMs={600} />
+          <div className="row">
+            <div className="col-sm-12">
+              <CountrySelect
+                addressDisabled={addressDisabled}
+                countries={countries.map(country => ({ name: country.name, displayName: country['display-name']}))}             
+                onChange={handleChange}
```

Comment:
This should call `refreshRegions` I think. Not seeing how that is being done here.

Reply:
I noticed that too, I'll push something that should address that.